utils/resb/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! `resb` is a utility crate of the [`ICU4X`] project for working with ICU
+//! resource bundle files.
+//!
+//! It comprises modules for reading and optionally writing [`binary`] `.res`
+//! files as well as optionally for reading [`text`] bundles.
+//!
+//! [`ICU4X`]: ../icu/index.html
+
+// https://github.com/unicode-org/icu4x/blob/main/docs/process/boilerplate.md#library-annotations
+#![cfg_attr(not(any(test, feature = "std")), no_std)]
+#![cfg_attr(
+    not(test),
+    deny(
+        clippy::indexing_slicing,
+        clippy::unwrap_used,
+        clippy::expect_used,
+        clippy::panic,
+        clippy::exhaustive_structs,
+        clippy::exhaustive_enums,
+        missing_debug_implementations,
+    )
+)]
+#![warn(missing_docs)]
+
+use core::fmt;
+
+extern crate alloc;
+use alloc::borrow::Cow;
+
+/// The `binary` module provides a `serde` deserializer for the binary resource
+/// bundle format as well as a means of writing resource bundles to the binary
+/// file format.
```

Comment:
Nit (optional): ICU4X style is to put these as `//!` comments in the module file.

Reply:
Addressed.